### PR TITLE
Clean up success spam, add labels to success/failure for better ID

### PR DESCRIPTION
**Pull request by dchw**

### CODE REVIEW

File: conslogging/conslogging.go [hunk 1]
```diff
@@ -130,17 +130,35 @@ func (cl ConsoleLogger) WithFailed(isFailed bool) ConsoleLogger {
...
 }
 
 // PrintFailure prints the failure message.
-func (cl ConsoleLogger) PrintFailure() {
+func (cl ConsoleLogger) PrintFailure(extra string) {
 	cl.mu.Lock()
 	defer cl.mu.Unlock()
-	cl.color(warnColor).Fprintf(cl.outW, "=========================== FAILURE ===========================\n")
+
+	cl.printBar(" FAILURE ", extra)
```
Comment (vladaionescu) [inline conslogging/conslogging.go:144]: Wouldn't this be printed with the incorrect color?

Comment (dchw) [reply conslogging/conslogging.go:144]: Shoot, victim of copy/paste

File: conslogging/conslogging.go [hunk 2]
```diff
@@ -130,17 +130,36 @@ func (cl ConsoleLogger) WithFailed(isFailed bool) ConsoleLogger {
...
-func (cl ConsoleLogger) PrintFailure() {
+func (cl ConsoleLogger) PrintFailure(msg string) {
 	cl.mu.Lock()
 	defer cl.mu.Unlock()
-	cl.color(warnColor).Fprintf(cl.outW, "=========================== FAILURE ===========================\n")
+
+	cl.printBar(warnColor, " FAILURE ", msg)
+}
+
+func (cl ConsoleLogger) printBar(color *color.Color, center, msg string) {
```
Comment (vladaionescu) [inline conslogging/conslogging.go:147]: Nit: Use `c *color.Color` to avoid clash with package name?

**Review by vladaionescu — COMMENTED** [review]
Can you show one or two screenshots of how this looks like?

**Comment by dchw** [comment]
> Can you show one or two screenshots of what this looks like?

Yep one sec, coming right up. VSCode debug window is monochrome only; so gotta do a build.

**Comment by dchw** [comment]
image

**Review by vladaionescu — APPROVED** [review]

**Comment by dchw** [comment]
@vladaionescu Swapped main to only print `SUCCESS`; and fixed up the hardcoded calls to the main success, since `--push` uses the same build functions now.

File: builder/builder.go [hunk 3]
```diff
@@ -110,18 +110,49 @@ func (b *Builder) MakeImageAsTarBuilderFun() states.DockerBuilderFun {
...
+				b.s.sm.SetSuccess("")
+			}
+		},
+		func() {
+			if opt.PrintSuccess {
+				b.s.sm.SetSuccess("--push")
+			}
+		},
+	)
+
```
Comment (dchw) [inline builder/builder.go:155]: Im not super hot here on this seeing it in the code review window; I could go to a factory like I had before, but still pump up the array...

File: builder/solver_monitor.go [hunk 4]
```diff
@@ -316,11 +317,12 @@ Loop:
...
 	if errVertex != nil {
-		sm.reprintFailure(errVertex)
+		sm.reprintFailure(errVertex, phaseText)
 	}
 	sm.mu.Lock()
-	if sm.success {
-		sm.console.PrintSuccess()
+	if sm.success && !sm.printedSuccess {
+		sm.console.PrintSuccess(phaseText)
+		sm.printedSuccess = true
```
Comment (vladaionescu) [inline builder/solver_monitor.go:325]: Should this also get `sm.lastOutputWasOngoingProgress = false`?

File: builder/solver_monitor.go [hunk 5]
```diff
@@ -419,9 +423,9 @@ func (sm *solverMonitor) PrintTiming() {
 		Printf("Total (real)\t%s\n", time.Now().Sub(sm.startTime))
 }
 
-func (sm *solverMonitor) reprintFailure(errVertex *vertexMonitor) {
+func (sm *solverMonitor) reprintFailure(errVertex *vertexMonitor, phaseText string) {
 	sm.console.Warnf("Repeating the output of the command that caused the failure\n")
-	sm.console.PrintFailure()
+	sm.console.PrintFailure(phaseText)
```
Comment (vladaionescu) [inline builder/solver_monitor.go:428]: Should this also get `sm.lastOutputWasOngoingProgress = false`?

**Review by vladaionescu — COMMENTED** [review]
Looks good! Just a remark

File: builder/solver_monitor.go (hunk 4 above)
Comment (dchw) [reply builder/solver_monitor.go:325]: Probably. Most cases we are ending up printing through SetSuccess now; so unlikely to get here first.

**Review by vladaionescu — APPROVED** [review]

**Pull request merged**